app/controllers/newflow/login_signup_controller.rb

```diff
@@ -326,6 +325,22 @@ def change_password
       end
     end
 
+    def exit_accounts
+      referrer_params = Addressable::URI.parse(request.referrer).query_values.to_h
+
+      if (return_to = referrer_params['r'])
```

Comment:
I would consider refactoring this to improve.    (1) not a great idea to assign variables inside of if statements.  (2) this method has high cyclometric complexity.  i encourage you to install and run flog on it https://blog.feabhas.com/2018/07/code-quality-cyclomatic-complexity/.   (3) does this method have unit tests covering all the if branches (one of the dangers of high cyclometric complexity, and bugs.

Reply:
Re: 1. I agree that it's uncommon and should be avoided if necessary but I insist that doing so in this case is alright — especially after the refactoring I just did/pushed. Requesting re-review. :) 
2. Just glanced at the rather long article but I get the point, I know what you mean. 
3. I just improved all of my code immensely, IMO. I'm hoping that'll suffice. Requesting re-review. Thanks!